### DISCUSSION TITLE
Possible issue with 'threshold' in rdFMCS.FindMCS()?

### DISCUSSION THREAD

**Discussion by jepdavidson** [rdkit/rdkit#5504] [Q&A]
Dear All,

I was recently using `FindMCS()` and was confused by what I was seeing.

`RDKit version = 2022.03.4 (py39h89e00b9_0, conda-forge)`

The docs state that *threshold=1.0* is the default for `FindMCS()`, and *threshold* is defined as:

> *property* **Threshold**
>     fraction of the dataset that must contain the MCS

However, what I am seeing seems more like *threshold=0.0* behaviour...  Here's a quick example:

```python
smis = ["C", "O", "N", "CC"]
mols = [Chem.MolFromSmiles(smi) for smi in smis]
mcs = rdFMCS.FindMCS(mols)
print(mcs.smartsString)
```
> [#6]

My reading of this is that the MCS code is saying that the maximum common substructure shared by **all** of the mols is a carbon atom(?!).  With a threshold of 1.0 I would expect **no** MCS(?).

I see the same behaviour if I explicitly set *threshold=1.0*.

```python
mcs = rdFMCS.FindMCS(mols, threshold=1.0)
print(mcs.smartsString)
```
> [#6]

So, my question is - am I misunderstanding the threshold parameter, or is there something else going on here?

Kind regards

James

**Comment by ptosco** [comment]
@jepdavidson Thanks for reporting. That looks like a bug independent of the `threshold` parameter. It only show up under certain circumstances, and I believe it has to do with how single-atoms seeds are handled.
I have filed this as #5504 and will add it to my ever-increasing list of MCS bugs I have been working on for a while, and hopefully fix it before next release.
